src/api_core.rs

```diff
@@ -35,15 +34,24 @@ pub struct ContractAddress(pub PatriciaKey);
 pub const MAX_STORAGE_ITEM_SIZE: u16 = 256;
 /// The prefix used in the calculation of a contract address.
 pub const CONTRACT_ADDRESS_PREFIX: &str = "STARKNET_CONTRACT_ADDRESS";
+
 /// The size of the contract address domain.
-pub static CONTRACT_ADDRESS_DOMAIN_SIZE: Lazy<StarkFelt> = Lazy::new(|| {
-    StarkFelt::try_from(PATRICIA_KEY_UPPER_BOUND)
-        .unwrap_or_else(|_| panic!("Failed to convert {PATRICIA_KEY_UPPER_BOUND} to StarkFelt"))
-});
+/// 0x800000000000000000000000000000000000000000000000000000000000000
+const CONTRACT_ADDRESS_DOMAIN_SIZE: StarkFelt = unsafe {
+    StarkFelt::new_unchecked([
+        8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0,
+    ])
+};
+
 /// The address upper bound; it is defined to be congruent with the storage var address upper bound.
-pub static L2_ADDRESS_UPPER_BOUND: Lazy<FieldElement> = Lazy::new(|| {
-    FieldElement::from(*CONTRACT_ADDRESS_DOMAIN_SIZE) - FieldElement::from(MAX_STORAGE_ITEM_SIZE)
-});
+/// CONTRACT_ADDRESS_DOMAIN_SIZE - MAX_STORAGE_ITEM_SIZE
+const L2_ADDRESS_UPPER_BOUND: FieldElement = FieldElement::from_mont([
+    18446743986131443745,
+    160989183,
+    18446744073709255680,
+    576459263475590224,
+]);
```

Comment:
once_cell does not support spin-based lock when compiled in no_std as stated here: https://docs.rs/once_cell/latest/once_cell/#faq

I tried lazy_static instead, but it has a weird way of supporting no_std. I support it by addition, instead of having it be the norm and std and addition.

Then I realized I could easily define those values as constant.